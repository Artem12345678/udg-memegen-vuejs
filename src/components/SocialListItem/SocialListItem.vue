<template>
  <a class="social__link" :href="url" target="_blank">
    <img :src="image">
  </a>
</template>

<script>
import instagram from '../../assets/instagram.svg';
import github from '../../assets/github.svg';

export default {
  name: 'SociaItem',
  props: ['url', 'icon'],
  computed: {
    image() {
      return this.icon === 'instagram' ? instagram : github;
    },
  },
};
</script>

<style lang="scss" scoped>
  .social__link {
    display: flex;
    width: 28px;
    height: 28px;
    transition: border-color .25s ease-in-out;
    text-align: center;
    border: 1px solid transparent;
    border-radius: 50%;
    align-items: center;
    justify-content: center;

    &:focus,
    &:active,
    &:hover {
      border-color: #fff;
    }
  }
</style>
