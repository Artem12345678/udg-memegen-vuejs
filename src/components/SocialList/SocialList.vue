<template>
  <ul class="social">
    <li v-for="(item, index) in links" :key="index" class="social__item">
      <SocialListItem :url="item.url" :icon="item.icon"/>
    </li>
  </ul>
</template>

<script>
import SocialListItem from '@/components/SocialListItem/SocialListItem.vue';

export default {
  name: 'SocialList',
  components: {
    SocialListItem,
  },
  data() {
    return {
      links: [
        {
          url: 'https://www.github.com/Artem12345678',
          icon: 'github',
        },
        {
          url: 'https://www.instagram.com/ystimenkoartem',
          icon: 'instagram',
        },
      ],
    };
  },
};
</script>

<style lang="scss">
  .social {
    display: flex;
    list-style-type: none;

    &__item {
      position: relative;
      z-index: 1;
      display: inline-block;
      margin-right: 10px;

      &:last-child {
        margin-right: 0;
      }
    }
  }
</style>
