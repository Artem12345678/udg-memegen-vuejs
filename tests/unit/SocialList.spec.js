import { shallowMount } from '@vue/test-utils';

import SocialList from '@/components/SocialList/SocialList.vue';
import SocialListItem from '@/components/SocialListItem/SocialListItem.vue';

describe('SocialList.vue', () => {
  let wrapper;

  const links = [
    {
      url: 'https://www.github.com/Artem12345678',
      icon: 'github',
    },
    {
      url: 'https://www.instagram.com/ystimenkoartem',
      icon: 'instagram',
    },
  ];

  beforeEach(() => {
    wrapper = shallowMount(SocialList);
    wrapper.setData({ links });
  });

  it('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders <SocialListItem/> components', () => {
    expect(wrapper.find(SocialListItem).exists()).toBe(true);
  });

  it('renders the right amount of <SocialListItem/> components', () => {
    expect(wrapper.findAll(SocialListItem).length).toBe(links.length);
  });
});
