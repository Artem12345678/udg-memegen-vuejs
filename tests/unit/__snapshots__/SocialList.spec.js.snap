// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialList.vue matches snapshot 1`] = `
<ul
  class="social"
>
  <li
    class="social__item"
  >
    <sociallistitem-stub
      icon="github"
      url="https://www.github.com/Artem12345678"
    />
  </li>
  <li
    class="social__item"
  >
    <sociallistitem-stub
      icon="instagram"
      url="https://www.instagram.com/ystimenkoartem"
    />
  </li>
</ul>
`;
